extra_data/tests/test_streamer.py

```diff
@@ -53,93 +53,32 @@ def compare_nested_dict(d1, d2, path=''):
                 raise ValueError('diff: {}{}'.format(path, key), v1, v2)
 
 
-@pytest.fixture(scope="session")
-def server_1():
-    server = ZMQStreamer(4444, maxlen=10, protocol_version='1.0')
-    yield server
-
-
-@pytest.fixture(scope="session")
-def server_2_2():
-    server = ZMQStreamer(5555, maxlen=10, protocol_version='2.2')
-    yield server
-
-
-@pytest.fixture(scope="session")
-def client():
-    client = Client('tcp://localhost:5555')
-    yield client
-
-
-class DummyFrame:
-    """Client._deserialize() now expects the message in ZMQ Frame objects.
-
-    TODO: avoid using a private method from extra_data for tests.
-    """
-
-    def __init__(self, data):
-        self.bytes = data
-        self.buffer = data
-
+@pytest.fixture(scope='function')
+def server(protocol_version):
```

Comment:
I believe it's possible to parameterise a fixture, if all of the tests which use it should try the same set of possible parameters.